eras/conway/impl/testlib/Test/Cardano/Ledger/Conway/Imp/GovSpec.hs

```diff
@@ -250,10 +251,72 @@ pparamUpdateSpec =
 
 proposalsWithVotingSpec ::
   forall era.
-  ConwayEraImp era =>
+  ( ConwayEraImp era
+  , InjectRuleFailure "LEDGER" ConwayGovPredFailure era
+  ) =>
   SpecWith (ImpTestState era)
 proposalsWithVotingSpec =
   describe "Proposals" $ do
+    describe "Failures expected after bootstrap phase" $ do
+      it "Fails predicate when treasury withdrawal has nonexistent return address" $ do
+        policy <- getGovPolicy
+        protVer <- getProtVer
+        registeredRewardAccount <- registerRewardAccount
+        netId <- use (impGlobalsL . to networkId)
+        unregisteredRewardAccount <- RewardAccount netId . KeyHashObj <$> freshKeyHash
+        deposit <- getsNES $ nesEsL . curPParamsEpochStateL . ppGovActionDepositL
+        anchor <- arbitrary
+        let mkTreasuryWithdrawal rewardAccount =
+              ProposalProcedure
+                { pProcDeposit = deposit
+                , pProcReturnAddr = registeredRewardAccount
+                , pProcGovAction =
+                    TreasuryWithdrawals (Map.singleton rewardAccount $ Coin 1_000) policy
+                , pProcAnchor = anchor
+                }
+            validTreasuryWithdrawal = mkTreasuryWithdrawal registeredRewardAccount
+            invalidTreasuryWithdrawal = mkTreasuryWithdrawal unregisteredRewardAccount
+        if HF.bootstrapPhase protVer
```

Comment:
I want the tests to run for both protocol versions, ideally, so that we are certain that not only are treasury withdrawals not allowed during the bootstrap phase, but also, the predicate failure is not reported for invalid return addresses. These failures should only be reported after the bootstrap phase so we don't break node-client protocol. When I was implementing these tests they did run for both protocol versions. Has something changed recently that makes these run only under `relevantDuringBootstrapSpec` now?

Reply:
It works like this: 
 - Gov. relevantDuringBootstrapSpec  is called only for protocol version 9 
 - Gov.spec (which itself calls relevantDuringBootstrapSpec) is called for version 10 
 
This means that  in order to have something be called for both version 9 and 10, we should include it in `relevantDuringBootstrapSpec` . Everything that is called in `spec` besides `relevantDuringBootstrapSpec` will only be called for 10 :

```
spec = do
  relevantDuringBootstrapSpec    -- called separately for 9  in `Imp` module 
  constitutionSpec   -- only 10
  proposalsWithVotingSpec  -- only 10. 
  votingSpec --..etc 
  policySpec
  predicateFailuresSpec
  unknownCostModelsSpec
```

Because you added the tests to `proposalsWithVotingSpec` , they are only invoked with 10. 
So I think it would be good to either add them to a function already called in `relevantDuringBootstrapSpec` or create a new one and call it from  `relevantDuringBootstrapSpec` 

Hope this makes sense, let me know if not.

Reply:
Well, my bad, I see now. I have moved the tests to the appropriate parts of `relevantDuringBootstrapSpec`.